File: python/meteroid/models/token_request.py
# this file is @generated
import dataclasses
import typing as t

from ..serialization import BaseModel


@dataclasses.dataclass
class TokenRequest(BaseModel):
    """Token request (from POST body, application/x-www-form-urlencoded)"""

    grant_type: str
    """Grant type: "authorization_code" or "refresh_token" """

    client_id: t.Optional[str] = None
    """Client ID (if not using HTTP Basic auth)"""

    client_secret: t.Optional[str] = None
    """Client secret (if not using HTTP Basic auth)"""

    code: t.Optional[str] = None
    """Authorization code (for authorization_code grant)"""

    code_verifier: t.Optional[str] = None
    """PKCE code verifier (for authorization_code grant with PKCE)"""

    redirect_uri: t.Optional[str] = None
    """Redirect URI (for authorization_code grant, must match the one used in /authorize)"""

    refresh_token: t.Optional[str] = None
    """Refresh token (for refresh_token grant)"""

File: go/token_request.go
// this file is @generated
package meteroid

// Token request (from POST body, application/x-www-form-urlencoded)
type TokenRequest struct {
	// Client ID (if not using HTTP Basic auth)
	ClientId *string `json:"client_id,omitempty"`

	// Client secret (if not using HTTP Basic auth)
	ClientSecret *string `json:"client_secret,omitempty"`

	// Authorization code (for authorization_code grant)
	Code *string `json:"code,omitempty"`

	// PKCE code verifier (for authorization_code grant with PKCE)
	CodeVerifier *string `json:"code_verifier,omitempty"`

	// Grant type: "authorization_code" or "refresh_token"
	GrantType string `json:"grant_type"`

	// Redirect URI (for authorization_code grant, must match the one used in /authorize)
	RedirectUri *string `json:"redirect_uri,omitempty"`

	// Refresh token (for refresh_token grant)
	RefreshToken *string `json:"refresh_token,omitempty"`
}
